library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +54,33 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  implicit class JSRichFuture[A](val self: Future[A]) {
+    /** Converts the Future to a JavaScript [[Promise]].
+     *
+     *  Attention! The nature of the [[Promise]] class, from the ECMAScript
+     *  specification, makes this method inherently un-typeable, because it is
+     *  not type parametric.
+     *
+     *  The signature of the `toJSPromise` method is only valid
+     *  <i>provided that</i> the values of `A` do not have a `then` method.
+     */
+    def toJSPromise: Promise[A] = {
```

Comment:
Exactly. Not that much voodoo required. Just a low priority on any and a high on Thenable :)